lib/power_action_utils.pm

```diff
@@ -282,6 +282,41 @@ sub assert_shutdown_and_restore_system {
             select_console($vnc_console);
         }
     }
+
+=head2 assert_shutdown_with_soft_timeout
+
+  assert_shutdown_with_soft_timeout({[,timeout => $timeout] [,soft_timeout => $soft_timeout] [, bugref => $bugref] [,soft_failure_reason => $soft_failure_reason]});
+
+Extending assert_shutdown with a soft timeout. When C<$soft_timeout> is reached,
+a soft failure is recorded with the message C<$soft_failure_reason>. After
+that, assert_shutdown continues until the (hard) timeout C<$timeout> is hit.
+
+This makes sense when a shutdown sporadically takes longer then it normally should take
+and the proper statistics of such cases should be gathered instead of just increasing
+a timeout.
+
+If C<$soft_timeout> is not specified, then the default assert_shutdown is executed.
+
+Example:
+
+  assert_shutdown_with_soft_timeout({timeout => 300, soft_timeout => 60, bugref => 'bsc#123456'});
+
+=cut
+
+    sub assert_shutdown_with_soft_timeout {
```

Comment:
something is fishy with the indention here, the sub should start at the first column, please check that. It looks like you define this function *within* assert_shutdown_and_restore_system, also see the additional closing `}` at the end